core/src/main/java/nu/yona/server/goals/service/GoalService.java

```diff
@@ -90,16 +91,30 @@ public GoalDTO updateGoal(UUID userID, UUID goalID, GoalDTO newGoalDTO, Optional
 		User userEntity = userService.getUserByID(userID);
 		Goal existingGoal = getGoalEntity(userEntity, goalID);
 
-		verifyGoalUpdate(existingGoal, newGoalDTO);
-
-		if (newGoalDTO.isGoalChanged(existingGoal))
+		if (newGoalDTO.getCreationTime().isPresent())
```

Comment:
This method has become more complex. Can't we just use the previous version and check on change in creation time?
Otherwise add a comment here explaining why this is an exceptional case.